### PR TITLE
[Sources] Use Immutable records

### PR DESCRIPTION
**Pull request by DavideDeCenzo**
Associated Issue: #4935 

Here's the Pull Request Doc
https://devtools-html.github.io/debugger.html/CONTRIBUTING.html#pull-requests

Here's the Debugger's Testing doc
https://docs.google.com/document/d/1oBMRxV8A2ag2t22YsQOxTdEv0mXKzIg0tggJjRkU1S0/edit#.
Feel free to improve it!

### CODE REVIEW

**Comment by jasonLaster** [comment]
- looks like this change needs to be rebased...
- also, i see some cases with `source.get("url")` which could be `source.url`

**Comment by DavideDeCenzo** [comment]
I've replaced  `source.get("url")` with  `source.url`  in one case but I've found also a case with `source.get("id")` and  one with  `get("tabs")`

File: src/reducers/sources.js [hunk 1]
```diff
@@ -306,25 +328,29 @@ export function getPendingSelectedLocation(state: OuterState) {
...
 
 function getSourceByUrlInSources(sources: SourcesMap, url: string) {
+
   if (!url) {
     return null;
   }
 
-  return sources.find(source => source.get("url") === url);
+  return sources.find(source => {
+    return source.get("url") === url;
```
Comment (jasonLaster) [inline src/reducers/sources.js]: shouldn't this be `source.url`?

**Comment by jasonLaster** [comment]
looks like it needs a rebase and jest is failing - https://circleci.com/gh/devtools-html/debugger.html/9919?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by DavideDeCenzo** [comment]
Now Jest should be ok but the travis build failed I think I've done a mess with this branch

**Comment by jasonLaster** [comment]
haha - how's it going @DavideDeCenzo would it help to do a rebase?

**Comment by DavideDeCenzo** [comment]
@jasonLaster Yes I don't think I do the rebase properly maybe I skip something

**Pull request closed without merging**